A39-xds-http-filters.md

```diff
@@ -0,0 +1,152 @@
+- [`HttpConnectionManager.http_filters`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L273)
+- [`VirtualHost.typed_per_filter_config`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/config/route/v3/route_components.proto#L142)
+- [`Route.typed_per_filter_config`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/config/route/v3/route_components.proto#L252)
+- [`WeightedCluster.ClusterWeight.typed_per_filter_config`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/config/route/v3/route_components.proto#L366)
+
+The `HttpConnectionManager.http_filters` field is an ordered list of
+filters.  In each entry, gRPC will look at the following fields:
+- [`name`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L833):
+  A logical name for the filter instance.  This name will be used when
+  applying overrides from the virtual host, route, or weighted cluster entry.
+  The logical instance name also allows configuring multiple instances of the
+  same filter, which can be useful for cases like authorization policies.
+- [`typed_config`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L838):
+  This field contains the configuration message for the filter.  gRPC
+  will contain a registry of filter implementations, keyed by the proto
+  message type used for its configuration; thus, when gRPC sees an entry
+  with a given proto message type, it will know which filter to
+  instantiate.  Note that as a special case, if the proto message type is
+  `udpa.type.v1.TypedStruct`, then gRPC will look inside that type's
+  [`type_url`](https://github.com/cncf/udpa/blob/cc1b757b3eddccaaaf0743cbb107742bb7e3ee4f/udpa/type/v1/typed_struct.proto#L38)
+  field to determine the actual type; this allows for third-party
+  plugins to encode their configurations as a `google.protobuf.Struct`
+  proto instead of using a concrete proto message type that the client
+  may not be able to handle.
+
+When gRPC receives the `HttpConnectionManager` proto in the LDS
+response, it will validate the list of filters as follows:
+- Every filter in the list must be of a known type, as identified by the
+  `type_url` field.  Any unknown config type will cause the Listener
+  resource to be NACKed.
+- Every filter in the list must have a unique instance name.  Any
+  duplicate instance names will cause the Listener resource to be NACKed.
+
+Note that gRPC will support configuring multiple instances of the same
+filter implementation, each with its own name and config.
```

Comment:
Core might have slightly more challenges comparing to Java/Golang, since Core is using service config to transport the filter config and Java/Golang can map each filter to an interceptor.

Reply:
That's true, but it's an implementation detail that doesn't need to be covered in the gRFC.  I'll deal with that as part of grpc/grpc#25310.